### DISCUSSION TITLE
STS throwing - >InvalidParameterValue<-expecting a policy to be set for user

### DISCUSSION THREAD

**Discussion by Bhagi33** [minio/minio#13433] [Q&A] [answered]
Hi Team,
I have configured the LDAP and set polices on user and also on groups.but unable to receive STS details.
PLease need your support in fixing this issue.
![image](https://user-images.githubusercontent.com/68953135/136774589-e97f3284-0104-412a-997f-4b49c69af816.png)

**Comment by klauspost** [comment]
@Bhagi33 It is not really clear that this is a MinIO issue. Are you using the latest version?

If you need help debugging your config/setup, please refer to our [subscription offer](https://min.io/pricing) where we can look at this in a confidential manner.

**Comment by Bhagi33** [comment]
Hi Klauspost,

 kubectl minio tenant info minio -n test-minio

Tenant 'minio', Namespace 'test-minio', Total capacity 80 GiB

  Current status: Initialized
  MinIO version: nsc-registry.infra.ericsson.se:5000/nsc/minio/minio:RELEASE.2021-09-24T00-24-24Z
  MinIO service: minio/ClusterIP (port 443)

  Console service: minio-console/ClusterIP (port 9443)

+------+---------+--------------------+---------------------+
| POOL | SERVERS | VOLUMES PER SERVER | CAPACITY PER VOLUME |
+------+---------+--------------------+---------------------+
| 0    | 2       | 4                  | 10Gi                |
+------+---------+--------------------+---------------------+

**Comment by klauspost** [comment]
STS had some fixes in the latest versions. Try upgrading and see if the issue still reproduces.

**Comment by Bhagi33** [comment]
i have upgraded tenant to minio/minio:RELEASE.2021-10-02T16-31-05Z. Tried testing for STS details. but still the issue is same.

**Comment by harshavardhana** [comment]
> Hi Team, I have configured the LDAP and set polices on user and also on groups.but unable to receive STS details. PLease need your support in fixing this issue. ![image](https://user-images.githubusercontent.com/68953135/136774589-e97f3284-0104-412a-997f-4b49c69af816.png)

It clearly says that you need to set the policy for the DN

```
mc admin policy set alias policyName user=userDN
```

**Comment by Bhagi33** [comment]
Hi HarshaVardhan,

Can you see the screenshot, i have already set policy to the user..You can see the policy name againist to the user under user list please.

**Comment by harshavardhana** [comment]
> Hi HarshaVardhan,
> 
> Can you see the screenshot, i have already set policy to the user..You can see the policy name againist to the user under user list please.

What you set is incorrect, check the response error in the XML. The error indicates what you need to do.

**Comment by Bhagi33** [comment]
Thanks Harsh,
I am able to get the Accesskey and Secret Key, but not able to login to Minio console..
why its happening like this.. 

kubectl logs minio-pool-0-0 -n minio-tenant
Error: LDAP server error: Unable to find user DN: User DN for A7NV8FIG9QWXEK9WENFO not found (*fmt.wrapError)
       3: cmd/sts-errors.go:58:cmd.writeSTSErrorResponse()
       2: cmd/sts-handlers.go:594:cmd.(*stsAPIHandlers).AssumeRoleWithLDAPIdentity()
       1: net/http/server.go:2046:http.HandlerFunc.ServeHTTP()

**Comment by donatello** [comment, marked as answer]
When logging in via the MinIO Console, you need to use the LDAP user (the short one, not the DN) and password, not the manually generated STS credentials, as the console does the STS credential generation for you.

**Comment by Bhagi33** [comment]
HI donatello,
Thanks dude :-)
ooops... Very confusing with STS flow in middle. Thanks for the information. I am able to login with ldap user.
